crates/vqueues/src/scheduler/vqueue_state.rs

```diff
@@ -152,39 +158,53 @@ where
     fn advance(&mut self, storage: &S) -> Result<(), StorageError> {
         loop {
             match self.reader {
-                Reader::NewWithRunning => {
-                    let mut it = storage.new_run_reader(&self.qid);
-                    it.seek_to_first();
-                    let item = it.peek()?;
+                Reader::New { already_running } if already_running > 0 => {
+                    let mut reader = storage.new_run_reader(&self.qid);
+                    reader.seek_to_first();
+                    let item = reader.peek()?;
                     if let Some(item) = item {
                         self.head = HeadStatus::Known(item);
-                        self.reader = Reader::Running(it);
+                        self.reader = Reader::Running {
+                            remaining: already_running,
+                            reader,
+                        };
                         break;
                     } else {
+                        error!(
+                            "vqueue {:?} has no running items but its metadata says that it has {already_running} running items",
+                            self.qid
+                        );
```

Comment:
This indicates that the metadata is no longer in sync, right? Should we treat this as a panic? Primarily wondering whether we can still make reliably progress if the meta is off (e.g. accounting for used tokens is wrong).

Reply:
Correct. It will panic in debug, but I tend to be a little conservative about panics in prod. No strong opinion here though. I can make this a harder error when I factor out the reader into its own file (which should happen soon).